internal/functions/efgs/utils/verification.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"fmt"
+	keyserverapi "github.com/google/exposure-notifications-server/pkg/api/v1"
+	"sort"
+	"strings"
+)
+
+// CalculateExposureKeysHMAC calculates HMAC of given keys. Copied from Verification server code.
+func CalculateExposureKeysHMAC(keys []keyserverapi.ExposureKey, hmacKey []byte) ([]byte, error) {
+	if len(keys) == 0 {
+		return nil, fmt.Errorf("cannot calculate hmac on empty exposure keys")
+	}
+	// Sort by the key
+	sort.Slice(keys, func(i int, j int) bool {
+		return strings.Compare(keys[i].Key, keys[j].Key) <= 0
+	})
+
+	// Build the cleartext.
+	perKeyText := make([]string, 0, len(keys))
+	altPerKeyText := make([]string, 0, len(keys))
+	calculateAlt := true
+	for _, ek := range keys {
+		perKeyText = append(perKeyText,
+			fmt.Sprintf("%s.%d.%d.%d", ek.Key, ek.IntervalNumber, ek.IntervalCount, ek.TransmissionRisk))
+		altPerKeyText = append(altPerKeyText, fmt.Sprintf("%s.%d.%d", ek.Key, ek.IntervalNumber, ek.IntervalCount))
```

Comment:
`altPerKeyText` is calculated but never used. `clearText` always contains joined `perKeyText`.

Reply:
Oh. Refactoring mistake.